src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -81,6 +81,13 @@ private void AddWellKnownTypes(IRootingServiceProvider rootProvider)
             {
                 rootProvider.AddCompilationRoot(stringType, "String type is always generated");
             }
+
+            var coreLib = _typeSystemContext.GetModuleForSimpleName("System.Private.Corelib");
+
+            // Adding for CPP codeGen
+            var eeTypeType = coreLib.GetKnownType("Internal.Runtime", "EEType");
+            rootProvider.AddCompilationRoot(eeTypeType, "Enables C++ Code Generation");
```

Comment:
(CoreLib actually didn't have the `Internal.Runtime.EEType` type up until a month or two ago and we didn't really touch EEType emission in CppCodegen for more than that - that's why I'm a bit suspicious.)